lib/txgh/github_api.rb

```diff
@@ -33,6 +33,24 @@ def create_ref(repo, branch, sha)
       client.create_ref(repo, branch, sha) rescue false
     end
 
+    def update_contents(repo, branch, content_map, message)
+      content_map.each do |path, new_contents|
+        branch = Utils.relative_branch(branch)
+        file = client.contents(repo, { path: path, ref: branch})
+        current_contents = Base64.decode64(file[:content])
+        current_sha = file[:sha]
+
+        new_sha = Utils.git_hash_blob(new_contents)
+        options = { branch: branch }
+
+        if current_sha != new_sha
```

Comment:
Would it be harmful to overwrite all the time? Would github figure out that it is a noop